lib/identity/index.js

```diff
@@ -0,0 +1,185 @@
+  const cryptoKeyPair = await id.crypto.deriveCryptoKey(
+    password,
+    id.config.realmId + id.config.realmName,
+    10000
+  )
+  const signingKeyPair = await id.crypto.deriveSigningKey(
+    password,
+    cryptoKeyPair.publicKey + cryptoKeyPair.privateKey,
+    10000
+  )
+  return { noteID, cryptoKeyPair, signingKeyPair }
+}
+
+function createClient(id, userId, storageConfig) {
+  const idConfig = id.config.clone({ userId: userId })
+  const storageClient = new id.StorageClient(storageConfig)
+  return new Client(idConfig, storageClient, id.crypto)
+}
+
+export default class Identity extends CryptoConsumer {
+  static get Client() {
+    return Client
+  }
+
+  static get Config() {
+    return Config
+  }
+
+  /**
+   * Abstract getter for a storage Client constructor function.
+   *
+   * When implementing this class, this getter must be overloaded. When called it
+   * should offer up a storage Client constructor function. Identity constructs
+   * storage clients as part of creating Identity clients.
+   *
+   * An additional instance level getter is also provided which allows fetching
+   * the storage client constructor in both static _and_ instance method
+   * contexts as `this.StorageClient`.
+   *
+   * @returns {Client} The storage Client constructor.
+   */
+  static get StorageClient() {
+    throw new Error(
+      'Implementing classes must overloaded the StorageClient method to provide a valid storage Client constructor.'
+    )
+  }
+
+  constructor(config) {
+    super()
+    this.config = config
+  }
+
+  /**
+   * Allows `this.StorageClient` syntax in instance methods.
+   *
+   * Gets the static StorageClient constructor available in the static class. By
+   * returning it as a getter `this.StorageClient` syntax is support in
+   * instance methods.
+   *
+   * @returns {Client} The storage Client constructor.
+   */
+  get StorageClient() {
+    // Use this.constructor to ensure we referencing the implementing class, not an interface class.
+    return this.constructor.StorageClient
+  }
+
+  async register(username, password, token) {
+    const cryptoKeys = await this.StorageClient.generateKeypair()
+    const signingKeys = await this.StorageClient.generateSigningKeypair()
+    /* eslint-disable camelcase */
+    const payload = {
+      realm_registration_token: token,
+      realm_id: this.config.realmId,
+      identity: {
+        realm_id: this.config.realmId,
+        name: username,
+        public_key: new PublicKey(cryptoKeys.publicKey),
+        signing_key: new SigningKey(signingKeys.publicKey)
+      }
+    }
+    /* eslint-enable */
+    const request = await fetch(this.config.apiUrl + '/v1/identity/register', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(payload)
+    })
+    const response = await checkStatus(request)
+    const json = await response.json()
```

Comment:
really want to see check status and parse JSON steps consolidated into a helper method that will allow us to dry out the repeated steps of constructing the request, making it, checking the status and parsing the response json.

Reply:
good point, this pattern is used frequently. I'll explore putting these together.

Reply:
added the helper and use it in this case -- other cases can adopt as they are modified.